ta/sims/ta_sims.c

```diff
@@ -44,13 +45,48 @@ static struct sims_bucket storage[TA_SIMS_MAX_STORAGE] = { {0} };
 
 static uint32_t counter;
 
+TEE_Result sims_open_ta_session(void *session_context, uint32_t param_types,
+				TEE_Param params[4])
+{
+	TEE_UUID *uuid;
+	TEE_Result res = TEE_SUCCESS;
+	TEE_TASessionHandle sess = TEE_HANDLE_NULL;
+	uint32_t ret_orig;
+	struct sims_session *ctx = (struct sims_session *)session_context;
+
+	if (param_types != TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+					   TEE_PARAM_TYPE_NONE,
+					   TEE_PARAM_TYPE_NONE,
+					   TEE_PARAM_TYPE_NONE))
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	uuid = TEE_Malloc(sizeof(*uuid), 0);
+	if (!uuid)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	TEE_MemMove((void *)uuid,
+		    params[0].memref.buffer,
+		    params[0].memref.size);
+
+	res = TEE_OpenTASession(uuid, 0, 0, NULL, &sess, &ret_orig);
+	TEE_Free(uuid);
+	if (res != TEE_SUCCESS)
+		return res;
+
+	ctx->sess = sess;
+	ctx->counter++;
+
+	return TEE_SUCCESS;
+}
+
 TEE_Result sims_open_session(void **ctx)
 {
 	struct sims_session *context =
 	    TEE_Malloc(sizeof(struct sims_session), 0);
 	if (context == NULL)
```

Comment:
add an empty line between variable definition and instructions.

Reply:
Done